File: KUASWifiAutoLogin/app/src/main/java/tw/edu/kuas/wifiautologin/callbacks/Reason.java
package tw.edu.kuas.wifiautologin.callbacks;

public class Reason {

    public static String dumpReason(int reason)
    {
        switch (reason)
        {
            case 1: return "登入頁面出現了系統錯誤！";
            case 2: return "沒有指定認證策略！";
            case 3: return "認證策略中自動增加指定的組不存在！";
            case 7: return "用戶已經被列入黑名單！";
            case 8: return "超出帳號最大登入數！";
            case 9: return "帳號綁定檢查失敗！";
            case 10: case 44: case 56: return "帳號不存在！";
            case 11: case 45: case 57: return "密碼不正確！";
            case 12: return "該帳號已經被凍結！";
            case 20:case 21:case 22:case 25: return "連接RADIUS伺服器時，發生了故障(" + reason +")！";
            case 24: return "無法連接到指定的RADIUS伺服器！";
            case 26: return "RADIUS伺服器回應數據不正確！";
            case 27:case 35: return "認證失敗，請檢查您的帳號及密碼 ";
            case 30: return "無法連接到指定的POP3伺服器！";
            case 31:case 32: return "連接POP3伺服器時，發生了故障(" + reason + ")！";
            case 33:case 34: return "POP3伺服器回應數據不正確(" + reason + ")！";
            case 40:case 42:case 43: return "連接LDAP伺服器時，發生了故障(" + reason + ")！";
            case 41: return "無法連接到指定的LDAP伺服器！";
            case 50: return "AD伺服器域名配置錯誤！";
            case 51:case 53:case 54: return "連接AD伺服器時，發生了故障(" + reason + ")！";
            case 52: return "無法連接到指定的AD伺服器！";
            case 55: return "AD伺服器的查詢密碼不正確！";
            case 60: return "登入失敗次數超出最大限制！";
            default: return "web_auth_error_" + reason;
        }
    }

}

File: Windows/KUAS WiFi/Reason.vb
Module Reason
    Public Function dumpReason(_reasonCode As Integer) As String
        Select Case _reasonCode
            Case 1
                Return "登入頁面出現了系統錯誤！"
            Case 2
                Return "沒有指定認證策略！"
            Case 3
                Return "認證策略中自動增加指定的組不存在！"
            Case 7
                Return "用戶已經被列入黑名單！"
            Case 8
                Return "超出帳號最大登入數！"
            Case 9
                Return "帳號綁定檢查失敗！"
            Case 10, 44, 56
                Return "帳號不存在！"
            Case 11, 45, 57
                Return "密碼不正確！"
            Case 12
                Return "該帳號已經被凍結！"
            Case 20, 21, 22, 25
                Return "連接RADIUS伺服器時，發生了故障(" + _reasonCode + ")！"
            Case 24
                Return "無法連接到指定的RADIUS伺服器！"
            Case 26
                Return "RADIUS伺服器回應數據不正確！"
            Case 27, 35
                Return "認證失敗，請檢查您的帳號及密碼。"
            Case 30
                Return "無法連接到指定的POP3伺服器！"
            Case 31, 32
                Return "連接POP3伺服器時，發生了故障(" + _reasonCode + ")！"
            Case 33, 34
                Return "POP3伺服器回應數據不正確(" + _reasonCode + ")！"
            Case 40, 42, 43
                Return "連接LDAP伺服器時，發生了故障(" + _reasonCode + ")！"
            Case 41
                Return "無法連接到指定的LDAP伺服器！"
            Case 50
                Return "AD伺服器域名配置錯誤！"
            Case 51, 53, 54
                Return "連接AD伺服器時，發生了故障(" + _reasonCode + ")！"
            Case 52
                Return "無法連接到指定的AD伺服器！"
            Case 55
                Return "AD伺服器的查詢密碼不正確！"
            Case 60
                Return "登入失敗次數超出最大限制！"
            Case Else
                Return "web_auth_error_" & _reasonCode
        End Select
    End Function

End Module
